selfdrive/common/params.h

```diff
@@ -12,35 +13,40 @@ class Params {
 
 public:
   Params(bool persistent_param = false);
-  Params(std::string path);
-
-  int write_db_value(std::string key, std::string dat);
-  int write_db_value(const char* key, const char* value, size_t value_size);
-
-  // Reads a value from the params database.
-  // Inputs:
-  //  key: The key to read.
-  //  value: A pointer where a newly allocated string containing the db value will
-  //         be written.
-  //  value_sz: A pointer where the size of value will be written. Does not
-  //            include the NULL terminator.
-  //  persistent_param: Boolean indicating if the param store in the /persist partition is to be used.
-  //                    e.g. for sensor calibration files. Will not be cleared after wipe or re-install.
-  //
-  // Returns: Negative on failure, otherwise 0.
-  int read_db_value(const char* key, char** value, size_t* value_sz);
-
-  // Delete a value from the params database.
-  // Inputs are the same as read_db_value, without value and value_sz.
-  int delete_db_value(std::string key);
-
-  // Reads a value from the params database, blocking until successful.
-  // Inputs are the same as read_db_value.
-  int read_db_value_blocking(const char* key, char** value, size_t* value_sz);
+  Params(const std::string &path);
+
+  // Delete a value
+  int remove(const char *key);
+  inline int remove(const std::string &key) {
+    return remove (key.c_str());
+  }
 
+  // read all values
   int read_db_all(std::map<std::string, std::string> *params);
-  std::vector<char> read_db_bytes(const char* param_name);
-  bool read_db_bool(const char* param_name);
 
-  std::string get(std::string key, bool block=false);
+  // read a value
+  std::string get(const char *key, bool block = false);
+
+  inline std::string get(const std::string &key, bool block = false) {
+    return get(key.c_str(), block);
+  }
+
+  template <class T>
+  std::optional<T> get(const char *key, bool block = false) {
+    std::istringstream iss(get(key, block));
+    T value{};
+    iss >> value;
+    return iss.fail() ? std::nullopt : std::optional(value);
+  }
+
+  inline bool getBool(const char *key) {
```

Comment:
Interesting that only `1` becomes true, not any other values. Did you find documentation for that? I couldn't find anything.

Seems that `1\n` also evaluates to true, but that's fine with me.

Reply:
fixed.

Reply:
fixed

Reply:
Yeah, maybe good to keep old behavior